feat: add logging to avante extension MCP tool calls

ksoichiro opened the pull request:
## Description

The avante extension does not log any information about the MCP tool, but I often want to know the details.
This pull request adds a logging to the `mcp_tool` function for avante extension.
Whenever the `use_mcp_tool` action is triggered, a log entry is created with details about the tool, server, and arguments being used.

## Screenshots

<!-- Add screenshots of the changes if applicable, to help visualize the change. -->
Before this change:
<img width="455" alt="before" src="https://github.com/user-attachments/assets/d284899a-9d31-4d48-9470-56d33cbcc674" />

After this change:
<img width="455" alt="after" src="https://github.com/user-attachments/assets/fe6ce13a-921f-413a-9fee-9299643d0d39" />

## Checklist

- [x] I've read the [contributing](https://github.com/ravitemer/mcphub.nvim/blob/main/CONTRIBUTING.md) guidelines and have adhered to them in this PR
- [ ] I've updated the README and/or relevant docs pages
- [x] I've run `make test` to ensure all tests pass
- [x] I've run `make format` to format the code
    - There was a change for the file which I didn't edit but I didn't include it in this PR.
- [ ] I've run `make docs` to update the vimdoc pages

ravitemer commented:
@ksoichiro Thank you for this PR. I didn't know `on_log` is for that purpose! It looks great. 

Can we also do the same for `access_mcp_resource` like `Accessing {uri} resource from {server}`. And currently we are using `vim.inspect(arguments)` which might fillup the whole buffer for calls like `write_file`, `replace_in_file` etc with the file contents. Can we transform the arguments such that the string fields of the arguments get's truncated to avoid filling up the buffer.

ksoichiro commented:
@ravitemer Thank you for your review!
I have added logging to `access_mcp_resource` as well, and changed to truncate string arguments with more than 80 characters (we can change the threshold if we need). While modifying, I realized there was an issue when truncating Japanese multibyte characters, so there is a bit complicated transformation to respect UTF-8 character boundaries.

Example output:

```
╭─  use_mcp_tool  succeeded
│   running tool
│   Calling tool `echo` on server `everything` with arguments: {
│     message = "This is another very long message that exceeds 100 characters in length. It dem... (truncated)"
│   }
╰─  tool finished
```

```
╭─  use_mcp_tool  succeeded
│   running tool
│   Calling tool `echo` on server `everything` with arguments: {
│     message = "これは日本語の長いメッセージです。多バイト文字のテスト... (truncated)"
│   }
╰─  tool finished
```

```
╭─  access_mcp_resource  succeeded
│   running tool
│   Accessing `mcphub://docs` resource from server `mcphub`
╰─  tool finished
```

ravitemer commented:
@ksoichiro Take a bow! Thanks for considering the utf-8 as well! Thanks again for your time.
